.github/workflows/codecov.yml

```diff
@@ -2,9 +2,12 @@
 name: Code Coverage Check
 
 on:
+  push:
+    branches:
+      - main
   pull_request:
-    branches: 
-    - main
+    types: [opened, synchronize, reopened]
```

Comment:
I believe this line is redundant. These activity types are what are set by default. https://docs.github.com/en/actions/using-workflows/events-that-trigger-workflows#pull_request.